compiler/src/dotty/tools/dotc/transform/SymUtils.scala

```diff
@@ -175,12 +175,16 @@ class SymUtils(val self: Symbol) extends AnyVal {
   def isEnumAnonymClass(implicit ctx: Context): Boolean =
     self.isAnonymousClass && (self.owner.name.eq(nme.DOLLAR_NEW) || self.owner.is(CaseVal))
 
-  /** Is this symbol defined locally (i.e. at some level owned by a term) and
-   *  defined in a different toplevel class than its supposed parent class `cls`?
-   *  Such children are not pickled, and have to be reconstituted manually.
+  /** Is this symbol defined locally (i.e. at some level owned by a term) so that
+   *  it cannot be seen from parent class `cls`?
    */
-  def isInaccessibleChildOf(cls: Symbol)(implicit ctx: Context): Boolean =
-    self.isLocal && !cls.topLevelClass.isLinkedWith(self.topLevelClass)
+  def isInaccessibleChildOf(cls: Symbol)(given Context): Boolean =
+    def isAccessible(sym: Symbol): Boolean =
+      sym == cls
+      || sym == cls.owner
+      || sym.is(Package)
+      || sym.isType && isAccessible(sym.owner)
+    !isAccessible(self.owner)
```

Comment:
```Scala
def foo: Unit = {
  object O { class A }
  class B extends O.A
}
```

According to the code, `B` is not accessible from `A`, is this intended?

Reply:
I would claim it _is_ accessible!

Reply:
Maybe I miss something here, I mean `B.isInaccessibleChildOf(A) = true`, that is the value I get from the compiler compiling the code.